blocks/graph.py

```diff
@@ -528,3 +531,107 @@ def apply_dropout(computation_graph, variables, drop_prob, rng=None,
         replacement.tag.replacement_of = variable
 
     return computation_graph.replace(replacements)
+
+
+def apply_batch_normalization(computation_graph, variables, gammas,
```

Comment:
I just remembered why it's not as straightforward as one may think: you need the actual size of `v` to initialize the shared variable. In your example, `v` is a symbolic variable, so `v.shape[-1]` is a symbolic variable as well.

At the very least, the user would need to provide the dimensionality of the non-normalized axes.